pyQuARC/code/datetime_validator.py

```diff
@@ -73,3 +75,47 @@ def compare(first, second, relation):
             "valid": result,
             "value": (str(first), str(second))
         }
+
+    @staticmethod
+    @if_arg
+    def validate_ending_datetime_against_granules(ending_datetime, collection_shortname):
+        """
+        Validates the collection EndingDatetime against the datetime of the last granule in the collection
+
+        Args:
+            ending_datetime (str): EndingDatetime string
+            collection_shortname (str): ShortName of the parent collection
+
+        Returns:
+            (dict) An object with the validity of the check and the instance
+        """
+        granules = requests.get(f'{CRM_URL}/search/granules.json?short_name={collection_shortname}&sort_key[]=-end_date').json()
+        last_granule = granules['feed']['entry'][0]
+        last_granule_endtime = last_granule['time_end']
+        
+        return {
+            "valid": ending_datetime == last_granule_endtime,
+            "value": (ending_datetime, last_granule_endtime)
+        }
+
+    @staticmethod
+    @if_arg
+    def validate_beginning_datetime_against_granules(beginning_datetime, collection_shortname):
```

Comment:
The previous method and this method are awfully similar. Can we not handle it someway else?